core/src/main/java/nu/yona/server/subscriptions/service/UserService.java

```diff
@@ -508,6 +508,10 @@ public User updateUser(UUID id, Consumer<User> updateAction)
 	{
 		User user = getUserEntityByIdWithUpdateLock(id);
 		updateAction.accept(user);
+		if (user.canAccessPrivateData())
```

Comment:
The assumption is that if the user anonymized is updated via the `updateAction`, this can only happen when the private data can be accessed (because the relation is encrypted). This I think is valid (it could be documented here).

Reply:
Like this?
```
			// The private data is accessible and might be updated
			// Let the UserAnonymizedService save it to the repository and cache it
```